src/helpers/WebhookMgr.ts

```diff
@@ -0,0 +1,175 @@
+import _ from "lodash";
+import { IDENTIFIER, CIRCLECI_URL } from "./config";
+import { getRecord } from "./getRecord";
+import { setExtensionFields } from "./setExtensionFields";
+
+export class WebhookMgr {
+  /**
+   * Handles Webhook
+   *
+   * @param headers
+   * @param payload
+   */
+  static webhookHandler = async ({ headers, payload }, { identifier, settings }) => {
+    console.log(`===== Received CircleCI ${identifier} =====`);
+    let fields: ICircleCIFields = {};
+    //Accepts only brances that follow Aha! naming convention
+    let reference = this.extractReference(payload.pipeline.vcs.branch);
+    let record;
+    //Get record name with branch / commit subject / commit description
+    if (reference) {
```

Comment:
This long block is very complicated, can we clean this up a bit to where it is not a nesting of if/ try/catch/ etc.  It would be nice to have a single call to extract a reference that passes or fails and then a single call to find the aha record that either passes or fails. The `extractReference` code can be moved into a separate function that takes the `payload.pipeline.vcs` and concatenates the branch and subject strings then uses a regex to find an reference numbers.

Reply:
I will think a bit more.

Reply:
Updated this part.